lumenlearning/lti-middleware
# L3-66 Troubleshoot integration between middleware and goldi SSO

## Description
Fixed issues related to differences between local and AWS environments as well as with the integration.
- Updated WebSecurityConfig to only send LTI requests through the LTI oAuth Filter. Canvas made several additional irrelevant requests that would break the LTI flow if they went through this filter because they would not contain the expected state.
- Added properties to the properties file so that Spring would include the Secure flag in the `set-cookie` header. Without this flag, the browser would refuse to set the jsessionid cookie, which is what Java uses to track the session, which is where the state is stored. Therefore, without the Secure cookie flag, the LTI state would get lost after the OIDC initiation.
- Used client side javascript to be able to perform a hard redirect POST to Goldilocks, ensuring that the UI displays properly.
- Updated unit tests accordingly and added instructions to the README on how to test in the AWS dev environment.

### Motivation and Context
These changes are necessary for the LTI integration between the middleware and Goldilocks to function.

### Fixes
[L3-66](https://lumenlearning.atlassian.net/browse/L3-66)

## How Has This Been Tested?
0. Register the middleware with Canvas and import the appropriate Common Cartridge into a new Canvas course.
1. Click on the links generated by the Common Cartridge.
2. Ensure that the study plan is visible, it is possible to navigate through the Dive In PBJ pages, and that it is possible to take quizzes.

---

## Checklist
